### ISSUE TITLE
release/public-v2: On MacOS, chgres_cube gives occasional segmentation faults when generating initial conditions with more than 1 processor

### ISSUE THREAD

**Issue by mkavulich** [NOAA-EMC/UFS_UTILS#275]
For the current head of the release/public-v2 branch, for MacOS Catalina 10.15.7 (and potentially other systems, though I have not seen any reports of this), chgres_cube will occasionally segfault when generating initial conditions with more than one processor. The problem is very hard to replicate reliably; sometimes running with a different number of cores will fix the problem, and sometimes recompiling with seemingly no changes will fix the problem. Running the same executable with a single core (mpirun -np 1) *always* completes successfully, which @climbfuji suggested is indicative of a potential race condition. 

Here is a sample log file of the failure:

```
 - READ TOTAL SOIL MOISTURE.
          17
 soilm    1.0004312992095947        1.9999997690320015E-002
 - CALL FieldScatter FOR INPUT TOTAL SOIL MOISTURE.
 - CALL FieldGather for INPUT SOIL TYPE.
 - CALL FieldGather for INPUT SOIL TYPE.
 - CALL FieldGather for INPUT SOIL TYPE.
 - CALL FieldGather for INPUT SOIL TYPE.
 - CALL FieldScatter FOR INPUT VEG TYPE.
 - CALL FieldScatter FOR INPUT VEG TYPE.
 - CALL FieldScatter FOR INPUT VEG TYPE.
 - READ VEG TYPE.
           9
 vgtyp    15.0000000       0.00000000    
 - CALL FieldScatter FOR INPUT VEG TYPE.
 - CALL FieldScatter FOR INPUT SOIL TYPE.
 - CALL FieldScatter FOR INPUT SOIL TYPE.
 - CALL FieldScatter FOR INPUT SOIL TYPE.
 - CALL FieldScatter FOR INPUT SOIL TYPE.
 - CALL FieldScatter FOR INPUT LANDSEA MASK.
 - CALL FieldScatter FOR INPUT LANDSEA MASK.
 - CALL FieldScatter FOR INPUT LANDSEA MASK.

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
 - CALL FieldScatter FOR INPUT LANDSEA MASK.
#0  0x10c470ebd
#1  0x10c4702ed
#0  0x10cb9aebd
#1  0x10cb9a2ed
#0  0x10e16eebd
#1  0x10e16e2ed
#2  0x7fff6f74b5fc
#2  0x7fff6f74b5fc
#2  0x7fff6f74b5fc
#3  0x108cef4e0
#3  0x1085cf4e0
#3  0x10a2d14e0
#4  0x1085dff9a
#4  0x10a2e1f9a
#4  0x108cfff9a
#5  0x10864c24d
#6  0x10858202f
#7  0x109d37f08
#5  0x10a34e24d
#6  0x10a28402f
#7  0x10ba39f08
#5  0x108d6c24d
#6  0x108ca202f
#7  0x10a457f08

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 96340 RUNNING AT vpn68-107.vpnclient.ucar.edu
=   EXIT CODE: 11
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
===================================================================================
YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Segmentation fault: 11 (signal 11)
```

**Comment by LarissaReames-NOAA** [comment]
Could you copy/paste or email me your namelist file (linked in the run directory as fort.41). A stack trace to the problematic line in input_data.F90 might be helpful as well if you get one.

**Comment by mkavulich** [comment]
@LarissaReames-NOAA I have attached the problematic namelist. 

Regarding the stack trace, what sort of flags should I compile with? Would setting DEBUG=YES get me a better trace to the error? As far as I can tell the utils are all compiled with the -g and -ftraceback flags but all I got out was those unhelpful hex values in the original message.

fort.41:
```
&config
    convert_atm = .true.
    convert_nst = .false.
    convert_sfc = .true.
    cycle_day = 15
    cycle_hour = 0
    cycle_mon = 6
    data_dir_input_grid = '/Volumes/d1/workdir/UFS/sharknadoadmin/ufs-srweather-app/experiment/test_CONUS_25km_GFSv15p2/2019061500/FV3GFS/for_ICS'
    external_model = 'GFS'
    fix_dir_input_grid = '/Volumes/d1/workdir/UFS/sharknadoadmin/data/'
    fix_dir_target_grid = '/Volumes/d1/workdir/UFS/sharknadoadmin/ufs-srweather-app/experiment/test_CONUS_25km_GFSv15p2/fix_lam'
    geogrid_file_input_grid = ''
    grib2_file_input_grid = 'gfs.pgrb2.0p25.f000'
    halo_blend = 10
    halo_bndy = 4
    input_type = 'grib2'
    lai_from_climo = .true.
    minmax_vgfrc_from_climo = .true.
    mosaic_file_target_grid = '/Volumes/d1/workdir/UFS/sharknadoadmin/ufs-srweather-app/experiment/test_CONUS_25km_GFSv15p2/fix_lam/C403_mosaic.halo4.nc'
    nsoill_out = 4
    orog_dir_target_grid = '/Volumes/d1/workdir/UFS/sharknadoadmin/ufs-srweather-app/experiment/test_CONUS_25km_GFSv15p2/fix_lam'
    orog_files_target_grid = 'C403_oro_data.tile7.halo4.nc'
    regional = 1
    sotyp_from_climo = .true.
    tg3_from_soil = .false.
    tracers = ''
    tracers_input = ''
    varmap_file = '/Volumes/d1/workdir/UFS/sharknadoadmin/ufs-srweather-app/src/UFS_UTILS/parm/varmap_tables/GFSphys_var_map.txt'
    vcoord_file_target_grid = '/Volumes/d1/workdir/UFS/sharknadoadmin/ufs-srweather-app/experiment/test_CONUS_25km_GFSv15p2/fix_am/global_hyblev.l65.txt'
    vgfrc_from_climo = .true.
    vgtyp_from_climo = .true.
/
```

**Comment by mkavulich** [comment]
Some "good" news, the error did not appear when I re-compiled everything (including NCEPLIBS-external and MPICH) with GNU 9.3.0 compilers (homebrew). So it is possible this error, whatever is causing it, is specific to GNU 10 or even GNU 10.2 compilers.

**Comment by mkavulich** [comment]
Resolved by #325, thanks @climbfuji!
